src/pages/execution/workflowTabs/gantt-chart/ConductorTimeline.tsx

```diff
@@ -0,0 +1,274 @@
+
+import {useEffect, useMemo, useState } from 'react';
+import { Button } from '@mui/material';
+import {
+    Bars,
+    Canvas,
+    Cursor,
+    GanttChart,
+    Highlight,
+    Series,
+    XAxis,
+    YAxis,
+  } from './';
+import {HighlightActions} from './HighlightActions';
+import { fontFamily, fontSizes, colors } from './internal/utils';
+import {Datum} from './types'
+
+const font = `${fontSizes.fontSize4} ${fontFamily.fontFamilySans}`;
+const [DO_WHILE, FORK_JOIN_DYNAMIC, FORK] = ["DO_WHILE", "FORK_JOIN_DYNAMIC","FORK"];
+const collapseTaskTypes = [DO_WHILE, FORK, FORK_JOIN_DYNAMIC];
+const [COMPLETED, FAILED, IN_PROGRESS, SCHEDULED, TIMED_OUT] = ["COMPLETED", "FAILED", "IN_PROGRESS", "SCHEDULED", "TIMED_OUT"]
+type ConductorTimelineProps = {
+  data: any, 
```

Comment:
data should be typed as `TaskResult[]`. However low priority as you might have to tinker with a lot to make it work. Save for separate PR later.